buildSrc/src/main/kotlin/BuildInfo.kt

```diff
@@ -80,6 +80,10 @@ open class BuildInfo(project: Project) {
 
   val isReleaseBuild: Boolean by lazy { java.lang.Boolean.getBoolean("releaseBuild") }
 
+  val isNativeArch: Boolean by lazy { java.lang.Boolean.getBoolean("nativeArch") }
```

Comment:
pass `gradlew ... -DnativeArch=true` to effectively pass [`-march=native`](https://www.graalvm.org/latest/reference-manual/native-image/overview/BuildOutput/#recommendations) to `native-image`